### ISSUE TITLE
Allow `cmd-click`ing indent guides to fold the code

### ISSUE THREAD

**Issue by JosephTLyons** [zed-industries/zed#12302]
### Check for existing issues

- [X] Completed

### Describe the feature

This comment had 33 upvotes

https://github.com/zed-industries/zed/issues/5373#issuecomment-1908654276

**Comment by JosephTLyons** [comment]
> ### Check for existing issues
> * [x]  Completed
> 
> ### Describe the feature
> This comment had 33 upvotes
> 
> [#5373 (comment)](https://github.com/zed-industries/zed/issues/5373#issuecomment-1908654276)
> 
> ### If applicable, add mockups / screenshots to help present your vision of the feature
> _No response_

Indent guides are in on main and will go out in Zed preview this Wednesday. We'll close this issue out shortly. Btw, this comment received 33 upvotes - I went ahead and made a new issue for it - feel free to upvote it:

- https://github.com/zed-industries/zed/issues/12302

**Comment by mcpeng** [comment]
I think this is a bit like Xcode's `Code folding ribbon`, but this feature will truly become practical because we don't need to move the cursor to the gutter area.
<img width="188" alt="code folding ribbon" src="https://github.com/zed-industries/zed/assets/2771843/5ecc8fc3-063c-41bf-b7e9-0e4e01a611ac">

**Comment by JosephTLyons** [comment]
There is one small bit of friction here in my mind and that is that once you click on the indent guide to fold the code, you must move your mouse to either the gutter or the `...` bit to unfold - this isn't a togglable action unless you move your mouse after the fact.  Not sure if that really matters, but it is something to consider.

**Comment by mcpeng** [comment]
Yes, what I'm describing is the folding operation. If you need to move the mouse to the gutter area at this time, it's like making your eyes walk a tightrope. For the unfolding operation, the precision requirement for mouse movement is not as high. Using the keyboard (such as in Vim mode) or clicking `...` is appropriate, allowing our eyes to flow within the code editing area.
